### PR TITLE
 feat(numericSelector): remove widget

### PR DESCRIPTION
**Pull request by francoischalifour**
This PR removes the widget NumericSelector as discussed in this PR: https://github.com/algolia/instantsearch.js/pull/3130#pullrequestreview-161232873.

### CODE REVIEW

File: docgen/src/guides/v3-migration.md [hunk 1]
```diff
@@ -308,6 +308,10 @@ With the redo button:
 </div>
 ```
 
+### NumericSelector
+
+Widget removed.
```
Comment (Haroenv) [inline docgen/src/guides/v3-migration.md]: should say "replaced by `numericMenu`" or something like that

Comment (francoischalifour) [reply docgen/src/guides/v3-migration.md]: It's not replaced but rather can be reproduced with NumericMenu. Let's write a guide later on?

Comment (Haroenv) [reply docgen/src/guides/v3-migration.md]: good for me, but should at least link to it IMO

Comment (francoischalifour) [reply docgen/src/guides/v3-migration.md]: I agree - will do another PR to review the whole migration guide. Let's move forward with this PR.

**Pull request merged**